centreon/www/include/configuration/configObject/host/DB-Func.php

```diff
@@ -2739,7 +2953,193 @@ function sanitizeFormHostParameters(array $ret): array
+    string $clientToken,
+    array $passwordTypeData,
+    CentreonUserLog $centreonLog,
+    CentreonRestHttp $httpClient
+): void {
+    try {
+        $url = $vaultConfiguration->getAddress() . ':' . $vaultConfiguration->getPort()
+            . '/v1/' . $vaultConfiguration->getStorage()
+            . '/monitoring/hosts/' . $hostId;
+        $centreonLog->insertLog(5, "Writing Host Secrets at : " . $url);
+        $httpClient->call($url, "POST", $passwordTypeData, ['X-Vault-Token: ' . $clientToken]);
+    } catch(\Exception $ex) {
+        $centreonLog->insertLog(5, "Unable to write host secrets into vault");
+
+        throw $ex;
+    }
+
+    $centreonLog->insertLog(
+        5,
+        sprintf(
+            "Write successfully secrets in vault: %s",
+            implode(', ', array_keys($passwordTypeData))
+        )
+    );
+}
+
+/**
+ * Update host table with secrets path on vault.
+ *
+ * @param VaultConfiguration $vaultConfiguration
+ * @param integer $hostId
+ * @param \CentreonDB $pearDB
+ * @throws \Throwable
+ */
+function updateHostTablesWithVaultPath(VaultConfiguration $vaultConfiguration, int $hostId, \CentreonDB $pearDB): void
+{
+    $path = "secret::" . $vaultConfiguration->getId() . "::" . $vaultConfiguration->getStorage()
+        . "/monitoring/hosts/" . $hostId;
+
+    $statementUpdateHost = $pearDB->prepare(
+        <<<SQL
+            UPDATE `host` SET host_snmp_community = :path WHERE host_id = :hostId
+        SQL
+    );
+    $statementUpdateHost->bindValue(':path', $path, \PDO::PARAM_STR);
+    $statementUpdateHost->bindValue(':hostId', (int) $hostId);
+    $statementUpdateHost->execute();
+}
+
+/**
+ * Get host secrets data from vault
+ *
+ * @param VaultConfiguration $vaultConfiguration
+ * @param integer $hostId
+ * @param string $clientToken
+ * @param CentreonUserLog $centreonLog
+ * @param CentreonRestHttp $httpClient
+ * @return array<string, mixed>
+ * @throws \Throwable
+ */
+function getHostSecretsFromVault(
+    VaultConfiguration $vaultConfiguration,
+    int $hostId,
+    string $clientToken,
+    CentreonUserLog $centreonLog,
+    CentreonRestHttp $httpClient
+): array {
+    $url = $vaultConfiguration->getAddress() . ':' . $vaultConfiguration->getPort() . '/v1/'
+        . $vaultConfiguration->getStorage() . '/monitoring/hosts/' . $hostId;
+    $centreonLog->insertLog(5, sprintf("Search Host %d secrets at: %s", $hostId, $url));
+    $hostSecrets = [];
+    try {
+        $content = $httpClient->call($url, 'GET', null, ['X-Vault-Token: ' . $clientToken]);
+    } catch (\RestNotFoundException $ex) {
+        $centreonLog->insertLog(5, sprintf("Host %d not found in vault", $hostId));
+
+        return $hostSecrets;
+    } catch (\Exception $ex) {
+        $centreonLog->insertLog(5, sprintf('Unable to get secrets for host : %d', $hostId));
+        throw $ex;
+    }
+    if (array_key_exists('data', $content)) {
+        $hostSecrets = $content['data'];
+    }
+
+    return $hostSecrets;
```

Comment:
Less variable, more explicit ?
```suggestion
    try {
        $content = $httpClient->call($url, 'GET', null, ['X-Vault-Token: ' . $clientToken]);
    } catch (\RestNotFoundException $ex) {
        $centreonLog->insertLog(5, sprintf("Host %d not found in vault", $hostId));

        return [];
    } catch (\Exception $ex) {
        $centreonLog->insertLog(5, sprintf('Unable to get secrets for host : %d', $hostId));
        throw $ex;
    }
    if (array_key_exists('data', $content)) {
        return $content['data'];
    }

    return [];
```